compiler/rustc_typeck/src/coherence/builtin.rs

```diff
@@ -606,7 +606,7 @@ pub fn coerce_unsized_info<'tcx>(tcx: TyCtxt<'tcx>, impl_did: DefId) -> CoerceUn
 
         // Finally, resolve all regions.
         let outlives_env = OutlivesEnvironment::new(param_env);
-        infcx.resolve_regions_and_report_errors(impl_did, &outlives_env);
+        infcx.check_region_obligations_and_report_errors(impl_did, &outlives_env);
```

Comment:
This is what fixes #53475 --- the rest of them I haven't looked into whether they can be made to ICE similarly, but I don't see any harm in changing those too.